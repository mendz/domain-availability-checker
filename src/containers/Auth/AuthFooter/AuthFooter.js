import React from 'react';
import PropTypes from 'prop-types';

import { IS_LOGIN, IS_SIGN_UP, IS_FORGOT_PASSWORD } from '../Auth';

import classes from './AuthFooter.module.css';

const Footer = ({ formType, changeFormType }) => {
  let footer = (
    <>
      <p>
        Don't have an account?{' '}
        <span
          className={classes.Link}
          onClick={() => changeFormType(IS_SIGN_UP)}
        >
          Sign Up
        </span>
        .
      </p>
      <p>
        or{' '}
        <span
          className={classes.Link}
          onClick={() => changeFormType(IS_FORGOT_PASSWORD)}
        >
          Forgot your password
        </span>
        ?
      </p>
    </>
  );

  if (formType === IS_SIGN_UP) {
    footer = (
      <p>
        Already have an account?{' '}
        <span className={classes.Link} onClick={() => changeFormType(IS_LOGIN)}>
          Log In
        </span>
        .
      </p>
    );
  }

  if (formType === IS_FORGOT_PASSWORD) {
    footer = (
      <>
        <p>
          After submit, please check your email for the rest password email.
        </p>
        <p>
          Already have an account?{' '}
          <span
            className={classes.Link}
            onClick={() => changeFormType(IS_LOGIN)}
          >
            Log In
          </span>
          .
        </p>
      </>
    );
  }
  return footer;
};

Footer.propTypes = {
  formType: PropTypes.string.isRequired,
  changeFormType: PropTypes.func.isRequired,
};

export default Footer;
