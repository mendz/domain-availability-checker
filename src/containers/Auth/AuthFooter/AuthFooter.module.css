.Link {
  color: var(--blue);
  cursor: pointer;
}

.Link:hover {
  text-decoration: underline;
}
