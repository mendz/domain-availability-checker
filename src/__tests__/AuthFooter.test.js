import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import AuthFooter from '../containers/Auth/AuthFooter/AuthFooter';
import {
  IS_LOGIN,
  IS_SIGN_UP,
  IS_FORGOT_PASSWORD,
} from '../containers/Auth/Auth';

describe('<AuthFooter />', () => {
  let wrapper;

  const initialProps = {
    formType: IS_LOGIN,
    changeFormType: jest.fn(() => {}),
  };

  beforeEach(() => {
    wrapper = shallow(<AuthFooter {...initialProps} />);
  });

  it('should match the snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should display the correct footer for form type="IS_LOGIN"', () => {
    expect(wrapper.text()).toContain(
      `Don't have an account? Sign Up.or Forgot your password?`
    );
  });

  it('should display the correct footer for form type="IS_SIGN_UP"', () => {
    wrapper.setProps({
      formType: IS_SIGN_UP,
    });
    expect(wrapper.text()).toContain(`Already have an account? Log In.`);
  });

  it('should display the correct footer for form type="IS_FORGOT_PASSWORD"', () => {
    wrapper.setProps({
      formType: IS_FORGOT_PASSWORD,
    });
    expect(wrapper.text()).toContain(`Already have an account? Log In.`);
  });

  it('should call props.changeFormType when click the span.Link', () => {
    wrapper
      .find('span.Link')
      .at(0)
      .simulate('click');
    expect(initialProps.changeFormType).toHaveBeenCalledTimes(1);
  });
});
