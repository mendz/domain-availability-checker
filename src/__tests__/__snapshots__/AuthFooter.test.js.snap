// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthFooter /> should match the snapshot 1`] = `
<Fragment>
  <p>
    Don't have an account?
     
    <span
      className="Link"
      onClick={[Function]}
    >
      Sign Up
    </span>
    .
  </p>
  <p>
    or
     
    <span
      className="Link"
      onClick={[Function]}
    >
      Forgot your password
    </span>
    ?
  </p>
</Fragment>
`;
